### DISCUSSION TITLE
Fluid grid gap

### DISCUSSION THREAD

**Discussion by edgarasben** [tailwindlabs/tailwindcss#3793] [Ideas]
Is there a way to make the grid gap size based on percentage and not pixel?

**Comment by browner12** [comment]
https://developer.mozilla.org/en-US/docs/Web/CSS/gap

Yes, but probably not with TW currently.

**Reply by edgarasben** [reply]
I would love to see this TW feature implemented in the future.

**Reply by dance2die** [reply]
In the mean time, you can [extend `gap`](https://tailwindcss.com/docs/gap#customizing) 

In your `tailwind.config.js` (code below adds 1~100 or add percentages you need one by one)

```js
const rangeFrom1To100 = [...Array(100).keys()].map((x) => ++x)
const gap = rangeFrom1To100.reduce((spacing, perc) => {
  const value = `${perc}%`
  spacing[value] = value

  return spacing
}, {})

module.exports = {
  theme: {
    extend: { gap },
  }
}
```

And specify the percentage

```html
  <div class="h-auto grid gap-5% grid-cols-2">
    <div class="border-4 border-pink-500">1</div>
    <div class="border-4 border-pink-500">2</div>
    <div class="border-4 border-pink-500">3</div>
    <div class="border-4 border-pink-500">4</div>
  </div>
```

Demo: https://play.tailwindcss.com/L5WMgaCrms

**Reply by edgarasben** [reply]
This is some elegant piece of magic!
